File: solutions/80/80.py
class Solution(object):
    def removeDuplicates(self, nums):
        count = 1   # Count of current element
        index = 1   # Index to place the next unique element

        for i in range(1, len(nums)):
            if (nums[i] == nums[i-1]):
                count += 1
            else:
                count = 1
                
            if (count <= 2):
                nums[index] = nums[i]
                index += 1
                
        return index

File: solutions/80/80.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }

        int count = 1;  // Count of current element
        int index = 1;  // Index to place the next unique element

        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1]) {
                count++;
            } else {
                count = 1;
            }

            if (count <= 2) {
                nums[index] = nums[i];
                index++;
            }
        }

        return index;
    }
};